### PR TITLE
Wrap ace3ds/r4sdhc.hk in an ifdef and rename r4sdhc_dualcore

### PR DESCRIPTION
**Pull request by jason0597**
First change is that I wrapped the entire `ace3dsplus.cpp` and `r4sdhc_dualcore.cpp` files in `#ifdef`s.

There is an issue with them not being in an `#ifdef`.
Suppose that e.g. you had flashcart_core and it was working fine, and then all of a sudden you get commits that support ace3dsplus, which is not yet fully finalized. After the ace3dsplus commits happen, you suddenly get some r4igold3ds fixes or something of the sort, fixes however that are stable and are able to be released.
Now, if you want to release those r4igold3ds fixes (or really any other fix), well, you got to wait until ace3dsplus is fully finished before you mark a release. You can't really grab the latest flashcart_core as it is, with the, still in testing phase, ace3dsplus code and mark a release with that. However, if you could simply have a way to grab the latest flashcart_core, _without_ having those testing phase devices be compiled, you could mark a release.

[I made a corresponding pull request to ntrboot_flasher's makefile for the `#ifdef` changes](https://github.com/ntrteam/ntrboot_flasher/pull/77)

==============================================================

Second change that I did is rename the `r4sdhc_dualcore.cpp` file to `r4igoldcc.cpp`
The `r4sdhc_dualcore.cpp` name is not a good name overall, as it causes confusion with the current `r4isdhc.cpp`. 

The reason why I renamed it to r4igoldcc and not say r4isdhc.hk/r4itt/r4ixds is because [the r4igold.cc issue](https://github.com/ntrteam/flashcart_core/issues/3) is the one that has had the most responses (359!), and seemed to be the main place where the discussion was happening
I also modified the corresponding lines (class declaration, logging, etc.) to match up with the filename change.

I also would like to note that I went ahead and wrote `getDescription()`. From all of the comments made on issue #3 , and from [this comment](https://github.com/ntrteam/flashcart_core/issues/3#issuecomment-345597365), I went ahead and wrote that the r4isdhc.hk/r4itt/r4ixds should be compatible as well. This is theoretical and not proven.

### CODE REVIEW

File: devices/ace3dsplus.cpp [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#ifdef TESTING
```
Comment (dark-samus) [inline devices/ace3dsplus.cpp]: this ifdef could be easily moved to ~line 586 (a similar change could be made to r4igoldcc.cpp as well); the "Ace3DSPlus ace3DSplus;" line is what adds it to the list of flashcarts, so it's way overkill to wrap the whole thing in ifdefs. In any case, I hate ifdefs, personally... in most cases anyway.

File: devices/r4igoldcc.cpp [hunk 2]
```diff
@@ -115,32 +117,39 @@ class R4SDHC_DualCore : Flashcart {
...
         return
                     "handsomematt, Normmatt, Kitlith,\n"
             "        stuckpixel, angelsl, et al.";
     }
+    const char *getDescription() {
+        return "Works with many various R4iGoldCC type flashcarts\n" 
+               " * R4i Gold CC (r4igold.cc)\n"
+               " * R4 SDHC Dual-Core (r4isdhc.hk)\n"
+               " * R4iTT 3DS (r4itt.net)\n"
+               " * R4i XDS 2014 (r4ixds.com)";
```
Comment (dark-samus) [inline devices/r4igoldcc.cpp]: This (r4ixds) actually doesn't appear to be the same cart, on a second look, so please remove it for now :)

File: devices/ace3dsplus.cpp [hunk 3]
```diff
@@ -584,3 +586,5 @@ class Ace3DSPlus : Flashcart {
 
 Ace3DSPlus ace3DSplus;
 }
+
+#endif
```
Comment (dark-samus) [inline devices/ace3dsplus.cpp]: If the ifdef is moved, this line needs to be moved from here to within the curly brace. Again, similar change for r4igoldcc

File: devices/ace3dsplus.cpp (hunk 1 above)
Comment (kitling) [reply devices/ace3dsplus.cpp]: I guess the difference is that this doesn't even have to compile and definately won't add any bloat to the executable if the whole thing removed, while it still has to compile (which honestly I find a *good* thing, makes it harder to accidentally make breaking changes) if we only remove that one line.

**Comment by kitling** [comment]
I agree with the rename of r4isdhc_dualcore to r4igoldcc. I agree that the old name was bad, it came from when I first looked at the flasher for a r4isdhc.hk cart, but later we supported the r4isdhc.com carts...

**Comment by jason0597** [comment]
commited changes

**Comment by angelsl** [comment]
I think the title should be "r4igold.cc" or just "R4i Gold". I'm inclined to the former, then we can rename R4iSDHC to r4i-sdhc.com or something like that, since all these carts don't really have a unique identifying name aside from the domain name, unlike Ace3DS or Acekard 2i.

**Comment by jason0597** [comment]
@angelsl there are so many ifs and buts everywhere that you have to expand a lot in the description

**Comment by angelsl** [comment]
Yes, I said "r4i-sdhc.com" specifically — note the TLD

**Comment by kitling** [comment]
LGTM, I'll probably merge tonight after adding a commit to change the titles a bit.

**Pull request merged**